src/app/shared/components/advertising-area/advertising-area.component.ts

```diff
@@ -0,0 +1,20 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-advertising-area',
+  templateUrl: './advertising-area.component.html',
+  styleUrls: ['./advertising-area.component.scss']
+})
+export class AdvertisingAreaComponent implements OnInit {
+
+  contentHeight: number;
+  contentWidth: number;
+  content: string;
+  preloadedPaddingBottom;
+
+  constructor() {}
+
+  ngOnInit() {
+      this.preloadedPaddingBottom = this.contentHeight / this.contentWidth * 100;
```

Comment:
the name preloadedPaddingBottom is not clear. Rename to placeholderRatio and place out of ngOnInit